### PR TITLE
Fix minor code quality issues

### PR DESCRIPTION
**Pull request by sauravsrijan**
Changes:

- Use `_` for unused variables
- Use `"""` for docstrings
- Add `.deepsource.toml` file to run continuous static analysis on the repository with DeepSource

This PR fixes some minor style issues and set up continuous static analysis on the repo with DeepSource. Upon enabling DeepSource, quality and security analysis will be run on every PR to detect 500+ types of problems in the changes — including bug risks, anti-patterns, security vulnerabilities, type-check issues, etc.
DeepSource is free to use for open-source projects, and is used by teams at NASA, Uber, Slack among many others, and open-source projects like ThoughtWorks/Gauge, Masonite Framework, etc.
To enable DeepSource analysis after merging this PR, please follow these steps:
- [Sign up](http://deepsource.io/signup) on DeepSource with your GitHub account and grant access to this repo.
- Activate analysis on this repo [here](https://deepsource.io/dashboard/gh/pydata/activate-repo/).
- You can also look at the [docs](https://deepsource.io/docs) for more details. Do let me know if I can be of any help!

### CODE REVIEW

File: .deepsource.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "xarray/tests/**"
+]
+
+[[analyzers]]
+name = "python"
+enabled = true
+
+  [analyzers.meta]
+  runtime_version = "3.x.x"
+  type_checker = "mypy"
```
Comment (crusaderky) [inline .deepsource.toml]: Isn't this redundant with the Azure CI job we already have?

**Comment by crusaderky** [comment]
If you merge from master you should get all green on the upstream-dev test now

**Comment by crusaderky** [comment]
Ok, I've played with it a bit and it's a hard -1 from me.

- I could not find a way to run it locally (correct me if I'm wrong); meaning all PRs will need to go through a million of trial-and-error commits and pushes
- Very heavily overlapping with what we already have (flake8, black, isort, mypy)
- Will trigger on issues that are explictly (made) ok for the other linter tools

**Comment by max-sixty** [comment]
Merging these as helpful coding changes (thank you @sauravsrijan ), even if we're not implementing DeepSource at the moment. We're open to reviewing in the future.

**Pull request merged**